### PR TITLE
Added ROADMAP.md

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Added initial ROADMAP.md
- Removed outdated NAS link in README.md

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- [ ] Parallel Hyperparameter Tuning
+- [ ] Distributed Model Training
+- [ ] Distributed Hyperparameter Tuning
+- [ ] Semi-supervised Learning
+- [ ] Enhanced Model Distillation
+- [ ] Online Inference Optimization
+- [ ] Meta-Learning
+- [ ] Improved Feature Type Inference
+- [ ] Co-variate Shift Detection
+- [ ] Co-variate Shift Correction
```
Comment (gradientsky) [inline ROADMAP.md:22]: Co-variate => [Covariate](https://www.statisticshowto.com/covariate/)

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+- [x] Make AutoGluon the most versatile AutoML framework via dedicated multi-modal image-text-tabular support ([paper](https://arxiv.org/abs/2111.02705)).
+- [x] Modularization of the various components of AutoGluon.
+- [x] Model Training Speed Optimizations.
+- [x] Model Inference Speed Optimizations.
+- [x] Model Quality Optimizations.
+- [x] Integration with [NVIDIA RAPIDS](https://developer.nvidia.com/rapids) for accelerated GPU training.
+- [x] Integration with [Intel sklearnex](https://github.com/intel/scikit-learn-intelex) for accelerated CPU training.
+- [x] Improved documentation and tutorials.
+
```
Comment (gradientsky) [inline ROADMAP.md:60]: + - [x] Training and Inference containers

**Comment by szha** [comment]
Job PR-1420-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1420/1/index.html

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+- [ ] Semi-supervised Learning
+- [ ] Enhanced Model Distillation
+- [ ] Online Inference Optimization
+- [ ] Meta-Learning
+- [ ] Improved Feature Type Inference
+- [ ] Co-variate Shift Detection
+- [ ] Co-variate Shift Correction
+- [ ] Exploratory Analysis
+- [ ] Model Interpretability
+- [ ] Model Uncertainty
```
Comment (smolix) [inline ROADMAP.md:25]: Probably also worth adding
- Model monitoring
- Calibration (conformal methods)
- Serving speed optimizations (this trade-off has come up a few times previously)
- Speed improvements for image and text tasks
- Semisupervised learning

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+In 2022, our top priorities are:
+
+- [ ] Windows OS Support
+- [ ] Python 3.9 Support
+- [ ] Time-Series Support
+- [ ] HuggingFace Integration
+- [ ] Improved Multi-modal Modeling
+- [ ] Advance SOTA in Stack Ensemble Research (Deeper Stack Ensembles)
+- [ ] Cloud Training & Deployment
+- [ ] Parallel Model Training
```
Comment (smolix) [inline ROADMAP.md:12]: Also worth putting on our list
- Dataset size scaling
- Improved data detectors for preprocessing
Should we discuss factoring out the preprocessor?

File: ROADMAP.md (hunk 3 above)
Comment (Innixma) [reply ROADMAP.md:25]: Good point, added:

```
- [ ] Model Monitoring
- [ ] Model Calibration (Conformal Methods)
- [ ] Image Model Inference Optimization
- [ ] Text Model Inference Optimization
```

Semi-supervised and online inference optimization are already listed.

File: ROADMAP.md (hunk 1 above)
Comment (Innixma) [reply ROADMAP.md:22]: fixed

File: ROADMAP.md (hunk 4 above)
Comment (Innixma) [reply ROADMAP.md:12]: added:

```
- [ ] Improved Feature Type Inference
- [ ] Improved Feature Preprocessing
- [ ] Improved Large-scale Data Handling (10M+ Rows)
- [ ] Refactor autogluon.features into Standalone Module
```

File: ROADMAP.md (hunk 2 above)
Comment (Innixma) [reply ROADMAP.md:68]: added

**Comment by szha** [comment]
Job PR-1420-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1420/3/index.html

**Review by gradientsky — APPROVED** [review]

**Pull request merged**